source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -859,13 +887,16 @@ const ShaderNode* ShaderGraph::getNode(const string& name) const
 void ShaderGraph::finalize(GenContext& context)
 {
     // Insert color transformation nodes where needed
-    for (const auto& it : _inputColorTransformMap)
-    {
-        addColorTransformNode(it.first, it.second, context);
-    }
-    for (const auto& it : _outputColorTransformMap)
-    {
-        addColorTransformNode(it.first, it.second, context);
+    if (context.getOptions().emitColorTransforms)
```

Comment:
Decision whether to emit is here.

Reply:
As a future improvement it would be nice to remove the old colortransform node insertions, and instead have shader code that performs the transformation immediately after texture sampling, inside the mx_image_color3/4 nodes, for the case where the OSL texture call does not support color transforms internally. Then the color transform is always done in the same place, and the most correct place.

Reply:
I think this is not hard to emit for non-OCIO generated code but there is a fair bit of customization currently needed to emit OCIO generated code and that is currently on a custom OCIO shader node including emitting support functions. Would work okay with OSL as functions can be embedded into other functions but unsure about something like GLSL.

Reply:
Yes this can definitely be delayed to another PR later.

I was imagining that we add a single call to a color transform function into the mx_image_color3/4 nodes, just after trhe texture sampling. And then the contents of this function can be customized depending on CM system used. The function can also be a no-op if CM is disabled or the specified colorspace requires no transformation. The CM system would then be encapsulated to a single entry point in the shader code, instead of having it scatted all over the shader as a result of all the various colortransform nodes being inserted in the graph.

In addition it would be the most correct place to perform the transform. since I think right now it's possible to create a nodegraph taking a filename, and in the graph have many math operations done on the result before a colortransform is applied on the output of this, which is incorrect.